### PR TITLE
`Development`: Migrate standard competencies admin import component

### PR DESCRIPTION
**Pull request by ole-ve**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context
Client migration to Angular 19. Includes standalone migration and use of inject(..)

### Steps for Testing
1. Import a standardized competency

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
  - The `AdminImportStandardizedCompetenciesComponent` has been updated to function as a standalone component, enhancing its modularity and independence.

- **Bug Fixes**
  - The `AdminImportStandardizedCompetenciesComponent` has been removed from the module's declarations, streamlining the admin functionalities while still allowing its use within the module.

- **Tests**
  - Updated test configurations to include additional modules and components for improved testing capabilities of the `AdminImportStandardizedCompetenciesComponent`.

### CODE REVIEW

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]

**Review by SimonEntholzer — DISMISSED** [review]
code changes lgtm

**Review by HawKhiem — APPROVED** [review]
Tested on TS3. Importing stardardized competencies works fine

image

File: src/main/webapp/app/admin/standardized-competencies/import/admin-import-standardized-competencies.component.ts [hunk 1]
```diff
@@ -76,12 +84,12 @@ export class AdminImportStandardizedCompetenciesComponent {
...
-        private adminStandardizedCompetencyService: AdminStandardizedCompetencyService,
-        private activatedRoute: ActivatedRoute,
-        private router: Router,
-    ) {}
+    private alertService = inject(AlertService);
+    private adminStandardizedCompetencyService = inject(AdminStandardizedCompetencyService);
+    private activatedRoute = inject(ActivatedRoute);
+    private router = inject(Router);
+
+    public constructor() {}
```
Comment (BBesrour) [inline src/main/webapp/app/admin/standardized-competencies/import/admin-import-standardized-competencies.component.ts:92]:
```suggestion
```

File: src/test/javascript/spec/component/standardized-competencies/admin-import-standardized-competencies.spec.ts [hunk 2]
```diff
@@ -15,14 +15,28 @@ import { HttpResponse } from '@angular/common/http';
...
+import { ArtemisMarkdownModule } from '../../../../../main/webapp/app/shared/markdown.module';
 
 describe('AdminImportStandardizedCompetenciesComponent', () => {
     let componentFixture: ComponentFixture<AdminImportStandardizedCompetenciesComponent>;
     let component: AdminImportStandardizedCompetenciesComponent;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ArtemisTestModule, NgbCollapseMocksModule],
+            imports: [
```
Comment (BBesrour) [inline src/test/javascript/spec/component/standardized-competencies/admin-import-standardized-competencies.spec.ts:30]: Consider coderabbits suggestion. I don't think we need all of these imports and declarations.

**Review by BBesrour — CHANGES_REQUESTED** [review]
code lgtm,
just small suggestions

**Review by florian-glombik — APPROVED** [review]
Code changes look reasonable

**Review by iyannsch — APPROVED** [review]
LGTM

**Review by HawKhiem — APPROVED** [review]
Tested again on TS3. Reapprove

Screenshot 2024-12-18 221110
Screenshot 2024-12-18 221123

**Review by Cathy0123456789 — APPROVED** [review]
Tested on TS3. Works as expected.

**Review by krusche — APPROVED** [review]

**Pull request merged**